xilem/examples/emoji_picker/main.rs

```diff
@@ -0,0 +1,173 @@
+            button("🔍-", |data: &mut EmojiPagination| {
+                data.size = (data.size + 1).min(5);
+            }),
+            // TODO: Expose that this is a "zoom in" button accessibly
+            button("🔍+", |data: &mut EmojiPagination| {
+                data.size = (data.size - 1).max(2);
+            }),
+        ))
+        .direction(Axis::Horizontal),
+        picker(data),
+        map_state(
+            paginate(
+                data.start_index,
+                (data.size * data.size) as usize,
+                EMOJI.len(),
+            ),
+            |state: &mut EmojiPagination| &mut state.start_index,
+        ),
+        data.last_selected
+            .map(|idx| label(format!("Selected: {}", EMOJI[idx].display)).text_size(40.)),
+    ))
+    .direction(Axis::Vertical)
+}
+
+fn picker(data: &mut EmojiPagination) -> impl WidgetView<EmojiPagination> {
+    let mut result = vec![];
+    // TODO: We should be able to use a grid view here, but that isn't implemented
+    // We hack around it by making each item take up their proportion of the 400
+    let dimensions = 400. / data.size as f64;
+    for y in 0..data.size as usize {
+        let mut row_contents = vec![];
+        let row_idx = data.start_index + y * data.size as usize;
+        for x in 0..data.size as usize {
+            let idx = row_idx + x;
+            let emoji = EMOJI.get(idx);
+            // TODO: Use OneOf2
+            let view: Box<AnyWidgetView<EmojiPagination>> = match emoji {
+                Some(emoji) => {
+                    let view = flex((
+                        // TODO: Expose that this button corresponds to the label below to accessibility?
+                        sized_box(button(emoji.display, move |data: &mut EmojiPagination| {
+                            data.last_selected = Some(idx);
+                        }))
+                        .expand_width(),
+                        sized_box(
+                            prose(emoji.name)
+                                .alignment(xilem::TextAlignment::Middle)
+                                .brush(if data.last_selected.is_some_and(|it| it == idx) {
+                                    // TODO: Ensure this selection indicator color is accessible
+                                    // TODO: Expose selected state to accessibility tree
+                                    palette::css::BLUE
+                                } else {
+                                    Color::WHITE
+                                }),
+                        )
+                        .expand_width(),
+                    ))
+                    .must_fill_major_axis(true);
+                    Box::new(view)
+                }
+                None => Box::new(flex(())),
+            };
+            row_contents.push(sized_box(view).width(dimensions).height(dimensions));
+        }
+        result.push(flex(row_contents).direction(Axis::Horizontal));
+    }
+
+    flex(result)
```

Comment:
This would be a good place to use a grid. The items can easily get out of alignment with the current implementation.